include/adc.h

```diff
@@ -27,119 +27,225 @@ extern "C" {
-	u8_t num_entries;
-	u8_t stride[3];
+	/**
+	 * Acquisition time.
+	 * Use the ADC_ACQ_TIME macro to compose the value for this field.
+	 * Particular drivers do not necessarily support all the possible units.
+	 * Value range is 0..16383 for a given unit.
+	 */
+	u16_t acquisition_time;
+
+	/** ADC reference selection. */
+	enum adc_reference  reference;
+	/** Channel voltage prescaling selection. */
+	enum adc_prescaling prescaling;
 };
 
 /**
  * @brief ADC driver API
  *
- * This structure holds all API function pointers.
+ * This is the mandatory API any ADC driver needs to expose.
  */
 struct adc_driver_api {
-	/** Pointer to the enable routine. */
-	void (*enable)(struct device *dev);
-
-	/** Pointer to the disable routine. */
-	void (*disable)(struct device *dev);
-
-	/** Pointer to the read routine. */
-	int (*read)(struct device *dev, struct adc_seq_table *seq_table);
+	int (*read)(struct device *dev,
+		    const struct adc_sequence *sequence);
+#ifdef CONFIG_ADC_ASYNC
+	int (*read_async)(struct device *dev,
+			  const struct adc_sequence *sequence);
+#endif
+	int (*channel_setup)(struct device *dev,
+			     const struct adc_channel_cfg *channel_cfg);
 };
 
 /**
- * @brief Enable ADC hardware
- *
- * This routine enables the ADC hardware block for data sampling for the
- * specified device.
+ * @brief Set a read request.
  *
  * @param dev Pointer to the device structure for the driver instance.
+ * @param sequence Structure specifying requested sequence of samplings.
  *
- * @return N/A
+ * @returns 0 on success, negative errno code otherwise.
  */
-__syscall void adc_enable(struct device *dev);
+__syscall int adc_read(struct device *dev,
+		       const struct adc_sequence *sequence);
 
-static inline void _impl_adc_enable(struct device *dev)
+static inline int _impl_adc_read(struct device *dev,
+				 const struct adc_sequence *sequence)
 {
 	const struct adc_driver_api *api = dev->driver_api;
 
-	api->enable(dev);
+	return api->read(dev, sequence);
 }
 
+#ifdef CONFIG_ADC_ASYNC
 /**
- * @brief Disable ADC hardware
- *
- * This routine disables the ADC hardware block for data sampling for the
- * specified device.
+ * @brief Set an asynchronous read request.
  *
  * @param dev Pointer to the device structure for the driver instance.
+ * @param sequence Structure specifying requested sequence of samplings.
  *
- * @return N/A
+ * @returns 0 on success, negative errno code otherwise.
  */
-__syscall void adc_disable(struct device *dev);
+__syscall int adc_read_async(struct device *dev,
+			     const struct adc_sequence *sequence,
+			     struct k_poll_signal *async);
 
-static inline void _impl_adc_disable(struct device *dev)
+static inline int _impl_adc_read_async(struct device *dev,
+				       const struct adc_sequence *sequence,
+				       struct k_poll_signal *async)
 {
-	const struct adc_driver_api *api = dev->driver_api;
+	const struct adc_driver_api *api = config->dev->driver_api;
```

Comment:
tiny left over from previous PR push

Reply:
Right. I haven't tried asynchronous part yet. That's why I haven't got any compile error for this.